payments-core/src/main/java/com/stripe/android/GooglePayJsonFactory.kt

```diff
@@ -300,54 +299,73 @@ class GooglePayJsonFactory constructor(
-
-        /**
-         * A unique ID that identifies a transaction attempt. Merchants may use an existing ID or
-         * generate a specific one for Google Pay transaction attempts. This field is required
-         * when you send callbacks to the Google Transaction Events API.
-         */
-        internal val transactionId: String? = null,
-
-        /**
-         * Total monetary value of the transaction.
-         *
-         * This field is required unless [totalPriceStatus] is set to [TotalPriceStatus.NotCurrentlyKnown].
+         * [TransactionInfo](https://developers.google.com/pay/api/android/reference/request-objects#TransactionInfo)
          *
-         * The value of this field is represented in the [smallest currency unit](https://stripe.com/docs/currencies#zero-decimal).
+         * @param currencyCode ISO 4217 alphabetic currency code.
+         * @param totalPriceStatus The status of the total price used.
+         * @param countryCode ISO 3166-1 alpha-2 country code where the transaction is processed. This
+         * is required for merchants based in European Economic Area (EEA) countries.
+         * @param transactionId A unique ID that identifies a transaction attempt. Merchants may use an
+         * existing ID or generate a specific one for Google Pay transaction attempts. This field is
+         * required when you send callbacks to the Google Transaction Events API.
+         * @param totalPrice Total monetary value of the transaction. This field is required unless
+         * [totalPriceStatus] is set to [TotalPriceStatus.NotCurrentlyKnown]. The value of this field is
+         * represented in the [smallest currency unit](https://stripe.com/docs/currencies#zero-decimal).
          * For example, when [currencyCode] is `"USD"`, a value of `100` represents 100 cents ($1.00).
+         * @param totalPriceLabel Custom label for the total price within the display items.
+         * @param checkoutOption Affects the submit button text displayed in the Google Pay payment sheet.
          */
-        internal val totalPrice: Int? = null,
-
-        /**
-         * Custom label for the total price within the display items.
-         */
-        internal val totalPriceLabel: String? = null,
+        @JvmOverloads
+        constructor(
+            currencyCode: String,
+            totalPriceStatus: TotalPriceStatus,
+            countryCode: String? = null,
+            transactionId: String? = null,
+            totalPrice: Int? = null,
+            totalPriceLabel: String? = null,
+            checkoutOption: CheckoutOption? = null,
+        ) : this(
+            currencyCode = currencyCode,
+            totalPriceStatus = totalPriceStatus,
+            countryCode = countryCode,
+            transactionId = transactionId,
+            totalPrice = totalPrice?.toLong(),
+            totalPriceLabel = totalPriceLabel,
+            checkoutOption = checkoutOption,
+        )
+
+        fun copy(
```

Comment:
We should deprecate, or mark this internal, or just not do it. At the very least note this should be removed in the next major version.

Reply:
Yep, deprecation coming to this one!